io/src/main/scala/sbt/internal/nio/FileCache.scala

```diff
@@ -137,8 +137,9 @@ private[nio] class FileCache[+T](converter: Path => T, globs: mutable.Set[Glob])
       case d            => (1 to d).foldLeft(Glob(path)) { case (g, _) => g / AnyPath }
     }
   }
+  private[this] val ceilingChar = (java.io.File.separatorChar.toInt + 1).toChar
   // This is a mildly hacky way of specifying an upper bound for children of a path
-  private[this] def ceiling(path: Path): Path = Paths.get(path.toString + Char.MaxValue)
+  private[this] def ceiling(path: Path): Path = Paths.get(path.toString + ceilingChar)
```

Comment:
```
scala> val xs = new TreeMap[Path, Int]()
xs: java.util.TreeMap[java.nio.file.Path,Int] = {}

scala> xs.put(Paths.get("/foo"), 1)
res1: Int = 0

scala> xs.put(Paths.get("/foo(2)"), 2)
res2: Int = 0

scala> xs.toString
res3: String = {/foo=1, /foo(2)=2}

scala> xs.subMap(Paths.get("/foo"), Paths.get("/foo0"))
res4: java.util.SortedMap[java.nio.file.Path,Int] = {/foo=1, /foo(2)=2}
```

Am I doing this right?

Reply:
Use tab `\u0009`?

Reply:
The idea behind the ceiling is that I want the Map.subMap(foo, ceiling(foo)) to include all of the possible children of foo. So if foo is the path "/foo", the subMap should include "/foo/bar". It's ok if the subMap includes some paths that are not children of foo (see the example all the way at the bottom). The ceiling is just a performance optimization so that we don't scan the whole map. The tab character would possibly be useable for the floor if we wanted to exclude "foo" from the submap.

```
scala> import java.nio.file._
import java.nio.file._                                                                         
                                                                                               
scala> val foo = Paths.get("foo")
foo: java.nio.file.Path = foo

scala> val bar = Paths.get("foo/bar")
bar: java.nio.file.Path = foo/bar

scala> val map = new java.util.TreeMap[Path, Int]
map: java.util.TreeMap[java.nio.file.Path,Int] = {}

scala> map.put(foo, 1)
res0: Int = 0

scala> map.put(bar, 2)
res1: Int = 0

scala> map.subMap(foo, Paths.get(foo.toString + '\t
'))
res3: java.util.SortedMap[java.nio.file.Path,Int] = {foo=1}

scala> map.subMap(foo, Paths.get(foo.toString + ('/'.toInt + 1).toChar))
res5: java.util.SortedMap[java.nio.file.Path,Int] = {foo=1, foo/bar=2}

scala> map.subMap(foo, Paths.get(foo.toString + ('\\'.toInt + 1).toChar))
res6: java.util.SortedMap[java.nio.file.Path,Int] = {foo=1, foo/bar=2}

scala> map.put(Paths.get("goo"), 2)
res7: Int = 0

scala> println(map)
{foo=1, foo/bar=2, goo=2}

scala> map.subMap(foo, Paths.get(foo.toString + ('\\'.toInt + 1).toChar))
res9: java.util.SortedMap[java.nio.file.Path,Int] = {foo=1, foo/bar=2}

scala> map.subMap(foo, Paths.get(foo.toString + ('\\'.toInt + 1).toChar))                      
res9: java.util.SortedMap[java.nio.file.Path,Int] = {foo=1, foo/bar=2}                         

scala> map.subMap(foo, Paths.get(foo.toString + ('/'.toInt + 1).toChar))
res10: java.util.SortedMap[java.nio.file.Path,Int] = {foo=1, foo/bar=2}

scala> map.subMap(foo, Paths.get(foo.toString + '\t'))
res11: java.util.SortedMap[java.nio.file.Path,Int] = {foo=1}

scala> map.put(Paths.get("fooo"), 3)
res12: Int = 0

scala> map.subMap(foo, Paths.get(foo.toString + ('/'.toInt + 1).toChar))
res13: java.util.SortedMap[java.nio.file.Path,Int] = {foo=1, foo/bar=2}

scala> map.put(Paths.get("foo."), 4)
res14: Int = 0

scala> map.subMap(foo, Paths.get(foo.toString + ('/'.toInt + 1).toChar))
res15: java.util.SortedMap[java.nio.file.Path,Int] = {foo=1, foo.=4, foo/bar=2}
```

Reply:
ok.